person.rb

```diff
@@ -0,0 +1,35 @@
+class Person
+  # Using attr_accessor to create the read and write methods for :name and :age.
+  attr_accessor :name, :age
+
+  # Using attr_reader to create the read method for :id.
+  attr_reader :id
+
+  # The constructor of the Person class with default parameters.
+  def initialize(name: 'Unknown', age: 0, parent_permission: true)
+    # Generation of a random identifier.
+    @id = generate_id
+
+    # Assignment of values ​​passed as parameters to instance variables.
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
```

Comment:
- [ ] Since the `age` parameter is not optional meaning when a person class is initialized providing an age is mandatory I'd suggest removing the default value of `0` you have assigned in the constructor.
- [ ] This comment also applies to the constructor of the `Student` and `Teacher` classes where age is assigned a default value of 0.